rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -684,7 +666,120 @@ public void onFailure(Throwable caught) {
         model.setNumPhysicalHosts(numPhysical);
         model.setNumVirtualHosts(numVirtual);
 
-        return ns;
+    }
+
+    private void recalculateStatistics() {
+        model.setNumBusy(0);
+        model.setNumConfiguring(0);
+        model.setNumDeploying(0);
+        model.setNumDown(0);
+        model.setNumFree(0);
+        model.setNumLost(0);
+        model.setNumToBeRemoved(0);
+        model.setNumLocked(0);
+        model.setNumDeployedNodeSources(0);
+        model.setNumUndeployedNodeSources(0);
+
+        for (NodeSource nodeSource : model.getNodeSources().values()) {
+            switch (nodeSource.getNodeSourceStatus()) {
+                case NODES_DEPLOYED:
+                    model.setNumDeployedNodeSources(model.getNumDeployedNodeSources() + 1);
+                    break;
+                case NODES_UNDEPLOYED:
+                    model.setNumUndeployedNodeSources(model.getNumUndeployedNodeSources() + 1);
+                    break;
+            }
+
+            for (Node node : nodeSource.getDeploying().values()) {
+                recalculateStatistics(node);
+            }
+
+            for (Host host : nodeSource.getHosts().values()) {
+                for (Node node : host.getNodes().values()) {
+                    recalculateStatistics(node);
+                }
+            }
+        }
+    }
+
+    private void recalculateStatistics(Node node) {
+        switch (node.getNodeState()) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)